### PR TITLE
CRISTAL-257: The edit button is presented even if the current user does not have edit rights

### PR DESCRIPTION
**Pull request by pjeanjean**
# Jira URL

https://jira.xwiki.org/browse/CRISTAL-257

# Changes

## Description

* Store whether or not the current user can edit the returned page.
* Hide the page actions if not.
* Show error on failed save.

## Clarifications

The changes in this PR are only related to the XWiki backend.

# Screenshots & Video

image

# Executed Tests

The test suite was run fully, and the new feature was tested with multiple rights combinations.

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  * N/A

### CODE REVIEW

File: core/backends/backend-xwiki/src/xwikiStorage.ts [hunk 1]
```diff
@@ -315,6 +316,14 @@ export class XWikiStorage extends AbstractStorage {
       // TODO: the syntax provided by the save is ignored and the content is always saved as markdown.
       body: JSON.stringify({ content, title, syntax: "markdown/1.2" }),
     });
+    if (!response.ok) {
+      const errorMessage = await response.text();
+      // TODO: Fix CRISTAL-383
```
Comment (manuelleduc) [inline core/backends/backend-xwiki/src/xwikiStorage.ts:321]: Maybe explain what's the issue with CRISTAL-383

File: core/backends/backend-xwiki/src/xwikiStorage.ts [hunk 2]
```diff
@@ -315,6 +316,14 @@ export class XWikiStorage extends AbstractStorage {
...
+      this.alertsServiceProvider
+        .get()
+        .error(`Could not save page ${page}. Reason: ${errorMessage}`);
+      throw new Error(errorMessage);
```
Comment (manuelleduc) [inline core/backends/backend-xwiki/src/xwikiStorage.ts:325]: Do we really need to throw an error if it is already handled by the alert service provider?

File: skin/src/vue/c-content.vue [hunk 3]
```diff
@@ -113,20 +113,25 @@ onUpdated(() => {
       <div class="doc-page-actions">
         <router-link
           :to="
-            currentPageRevision
+            currentPageRevision || !currentPage?.canEdit
```
Comment (manuelleduc) [inline skin/src/vue/c-content.vue:116]: Wouldn't it be better to simply hide the edit button when the user does not have the right to edit?

File: core/backends/backend-xwiki/src/xwikiStorage.ts (hunk 2 above)
Comment (pjeanjean) [reply core/backends/backend-xwiki/src/xwikiStorage.ts:325]: Actually, yes, we need it to update the editor status (that shows "Unsaved" only if it catches an error).

Comment (manuelleduc) [reply core/backends/backend-xwiki/src/xwikiStorage.ts:325]: ok, maybe add a comment to explain that, so that nobody removes it by accident.

Comment (pjeanjean) [reply core/backends/backend-xwiki/src/xwikiStorage.ts:325]: Good point.

**Pull request merged**